community-operators/postgresql/postgres-operator.v3.5.0.clusterserviceversion.yaml

```diff
@@ -0,0 +1,351 @@
/lNEcEHAvGGfM1/8wweUzhBAF1pmIswaw2rVrb/4t3/BNN91kNQwHRADGadOm2XSq559/3mqckfy4gBVm+vvuu9/0V0GUxjLTO8ANqOMvB+Cc7yJORF9iBVi7rmhlOWd7Fk46XbVte5i57LLL5NtvYQuvALQch/F552/AkC5VZ5xxho5fx5b+RJNKhDivm2662fRTBDq+bq9EpzgEmFr7gv68gjmaGcLHyE9H2hgGhCyfvHEgJMgp9Wvf9feyF9cFHlPt7ZtvvilWwRksVuTNn3HGmXHvcy9z27s3397L1Hn3ev9wLjxnlBL2TAj7e3aFtPTK/v3mmW+lsaG6B5fPC7y197xD15jRo62Gth/Qez5SxmzffbwehEtLY27+MdPAgY2LjRnzjAmqQxqlHuNp5fizL1H+86uvvmKjrgFDZ/FAK6fi211qNjJ+/Pioec815TukVejv//AHG/DlBHkxDmbD2rVqS0OtaKN3Y50xgsKq1Wui5rDbfPU6tVWZrrHtJe6ea7RxmQPA3rVrF0Pw0M8//2ytDdG2D/+eGur3/uY+u0iG/xbtbzQqZ2F1eBltW+d7wGj+/AXmrzLrExdw8smnWH+s1/2dcX7t7wh8aCaAYDzh0eEVPMYd0qbNYVojR9s8f+e3RN65f3k1y8mxdRYS2TfatnSi4/7lXvdCCOIIJ+Plblqk1DzPtFNlYbHS2lrv3o7leWx/w1RyIGDKmNHBKofONqNfs+lJ+wG9w2XlAvlGIY7Bnqk8oj9WKXBgl4Lepg4xwbHPGbNOD3KwIAgy4mRYxNCiaZ4yYMAAWyvcAQ+iqUeN+trcfvvt5iXVGd+0aWNEHy6FZTCv//GPfzQX7usE5owBqNVSL3O6puEz1uFiEoC2WmCOmdxNjMdYmFH5jcAnFjjnOO5tY31moevQoYMVBPCrI6zEo2O6d7f59ZGaokTalzkhIOGuIKDJK+Fr/0C94u+RD5Sc/j59+1mrgtdF3Otxfi3b4bIoq3vOnZUR79y5drQnpaQx7pxk3TO2n7vGwtXiWLniHTvW7wgINo1Nz4DX8bhvjlLgJl37wp+nqMeiOuSubSZQQY1mKA3rU+ZyIBD4Mbgr7ztVjLOL/H5AP/a28oG8nddq9SdtzacDjQMEteTJXDbnKxMc8x9jflE2A5GrMYiFCw36nHPOMW+/PUia4EkWHFk4iKieqBSrB5UbDchTfMNN7MtigaZ7wQUXmocfGWCbszTQ32hDaEVV1L+85j4gZ0Fln3iEVoVmHh5kx5gA4/x588zCRQtNi+bNozYpiXcMfmcu5I5XrlzFBstFrxQWMK3l///HP/8loecQT+fAYku7VzTzMmWyvEzH+mvnzJljBaK//vWvNmOgb79+Nhoef2iyhHAET0lhSgTUkj1epu3H+XPXIUgmKhRx3XE/jRo12vIv0XPbK42asrXEjyAwp4JoL1tBAXxeis1wPO5zGhxheieIFX54IsrCCtgDdVr6bVY9Max0NtL1XRHMrjnMjP+fSn2qjEDBNHYtLaM2CIcV/O1/SJIDemAKPTRaTiyOxQezpA6INL1XwL14kjHTPzLBdYv0INprG3M4/MqktzysCPY+ygtndkjwBKGhGZKGxgLgBlaAikURrRiTJILAKdI+MFHyG4CBgFBJC06iiycLDW0y165dVygIzppNNbnx339vpk2fZnqr+cUJJxxvz419eBH5zULHHJxe0l6EB7bp0+caHXONjRkgDzmcKLv6pz/9Wf73pp7AnPNmAaXtqxdCeILnw4ePkIXk72b69OlWICKYikYf0TqrxRs7xJM9tmkHVfNaSOP8P6UN/hqJNL/sHdk2loJ7JBE666xeZvDgQeZaxYKQo879mAgtVUDjyJEjbXpjNLN/vHvVDcLO/ZJIWh5Wrb5qgDRixAg9019q+h5BfeVPxswfbUzny9R307uVKRH++NsWjwMK+T3MCLudUfYDena18mb7nubOD/67Rw4AqACofd8TqoFOnrejHZdT8FSFaiEpNyC+0x40oEXFvhIF+X3CAto4x+MYy6YZM+sTE1wzV8Cu48chAAdpnWYqv1dXMIqZkPY1c+ZMdWsaJmAZZhao6AnkgDT7oCW2bNlKpUlPEKCeaLVWAJwXvnGn+EWcw0f8mUWKqmcEqRH8w98IF5ifiRT+Quk3zPePqlDH8Vjg2I7ft27basvCOoseAWiYurEOsJDFWyxZZG+++WZ7zm+99VYhTQw3wkPKv+/QsYMnAQUhiYA/eBGPmC9Bf3PnzlMTm2etSZRoZOZDlbHLr7jcBvHFG8f9O2OSEYCAgDA2eNAg2zK2SdNm5k+qM59qcngbMryEUhGdYzAX5+V8V1rvzIOo9/Ky8Dg91hOZC9URP/nkE5v1QUU5t5AbaRz4wrPD/QBvFi1cYC1cbdq0sX8H9OyzDdea8rNs6+wT4G/95vCW8bGWIZTwXEAUzqlcqbLd372d/THKP2SUUC9i8uQpCvZbF2WrsK93b1fr5MkmUK+1MU2PQt0P28D/s9Q5EDTNDdi9j/YBejBQfs/n9QQHkRstO1v77/s4IGAFTHXDm1WzjVkqUF09xwQ3SsPL21mUS7QFrdrABOq2Mqa2ZKaaTfR3feV6ypQa0G8OuIc/MFqIrDTNO35wRZ+a7etV2U055Bt+EZhPN8ENSzwBOY0eMFkfcYR6dT9wv62hTo4qNcXHSiNfsnhxaIHRdviJKwv0mwkIqIJ2hNp2ElyDed1ZeCoLNKtVq7pv0UruQWeBokY7lc9YJAFxFi9SfqgFP2bMaAXWdbe+++YysUPsw+JGgZVI6Wz4TMnXZgHH7I3wAsjHWvhY7G699VYb+Q5POAaWhltvu93m4WORiEWMjWWAtCcv3b4YHxM/oPvPf/5Ti+xkC34s7OTsX3nVVbZzW6xjun+zPBHfKPc5YsRw854C6bA28D2C2IABA2yKoHufRD47vOOdF/Msq5rp5cqGsgn4TDQ5fAakHMECgYtrRala5lKaxL0
```

Comment:
This apiGroup should be removed; otherwise this grants all permissions to the service account.

Reply:
@jeremylinlin The line of code you comment on defines an icon and not a permission.

Reply:
@jeremylinlin it looks like github renders this wrong in the conversation thread. @jmckind If this operator is deployed in a namespace where the operatorGroup include `AllNamespaces` the operator will have cluster permissions for api.

Reply:
I assume we are referring to the following:

`        - serviceAccountName: postgres-operator
          rules:
          - apiGroups:
            - "*"
            resources:
            - "*"
            verbs:
            - "*"`

This was taken from the existing Role created by the authors of the operator. I need to dig in further to see why this was setup this way. As this operator was created well before the current operator pattern was in place, with the intention of living in a single namespace, it may just need updating to be more restrictive. I need to make sure that changing this does not break the operator.